SSA/Projects/FullyHomomorphicEncryption/Syntax.lean

```diff
@@ -105,11 +90,30 @@ def ROfZComputable_impl (z : ℤ) : R q n :=
   }
   R.fromPoly p
 
-@[implemented_by ROfZComputable_impl]
-axiom ROfZComputable_stuck_term (q n : Nat) (z : ℤ) : R q n
+/-#
+Explanation of what's going on:
+- We want to have a `cstComputable : Z -> R`, which achieves three properties:
+1. It is computable, since the meta code needs it to be computable for `[fhe_icom|..]` to elaborate.
+2. It is equal to the noncomputable definition `(coe z : R q n)`
```

Comment:
> It is equal to the noncomputable definition `(coe z : R q n)`

Does it have to be def-eq? My guess would be no, since this equality used to be an axiom, so I get the feeling we can get rid of the `implemented_by` as well, if we can prove that `ROfZComputable_impl` is equal to `coe`.